src/doc/doc-printer.js

```diff
@@ -479,8 +479,13 @@ function printDocToString(doc, options) {
                     );
                   }
                 }
-
-                out.push(newLine + ind.value);
+                // make sure there is no newline after the it block, issue #3791
```

Comment:
Sorry, we should point you to similar PRs for guiding --> #3735.

The proper fix should be here:

https://github.com/prettier/prettier/blob/f2f239011454b2042f29333378e39d6a0a75de70/src/language-js/printer-estree.js#L5307-L5327

You'll have to add a case for `async(() => {})`, and you can see [astexplorer](http://astexplorer.net/#/gist/1622faf747bd792b8a005217161a23e0/2f57748ebce91326205900a84601fb0efc1b5712) for how its AST looks like.

Reply:
And see also #4010.

Reply:
thanks @ikatyang  I will fix the PR.